kolibri/core/assets/src/exams/utils.js

```diff
@@ -107,12 +133,22 @@ export function convertExamQuestionSources(exam, extraArgs = {}) {
   if (data_model_version === 1) {
     return annotateQuestionsWithItem(convertExamQuestionSourcesV1V2(exam.question_sources));
   }
+  // For backwards compatibility. If you are using V3, use the convertExamQuestionSourcesToV3 func
+  if (data_model_version === 3) {
+    return revertV3toV2(exam.question_sources);
+  }
+
   return annotateQuestionsWithItem(exam.question_sources);
 }
 
 export function fetchNodeDataAndConvertExam(exam) {
   const { data_model_version } = exam;
-  if (data_model_version >= 2) {
+  if (data_model_version >= 3) {
+    /* For backwards compatibility, we need to convert V3 to V2 */
+    exam.question_sources = revertV3toV2(exam.question_sources);
```

Comment:
As above!

Reply:
Our approach was to basically just extend the interface and do what we needed to to avoid breaking everywhere else in the app that relies on the V2 data structure. 

So until we update all parts of the app dependent on the old version to use V3 instead this ought to keep things working but eventually we can remove these bits.